I7 ekf

ajaysinghchhonker opened the pull request:
modular breakdown of code. Created EKF function which takes motion & sensor models along with the corresponding linearized model. Created motion & sensor model for a 3 wheeled omnibot.
An example "EKF_omnibot_example.m" is created for visualization of the work.

ajaysinghchhonker committed:
Delete ekf.m

deleting for creating a function named EKF

ajaysinghchhonker committed:
1st commit

Modular breakdown of the code

mcoteCPR commented on matlab_simulation/code v1.0/ME597-5-ExtendedKalmanFilter/EKF.m:
```diff
@@ -0,0 +1,24 @@
+    Ht = linearize_sensor_model(mu);
+    
+    % Prediction
+    mup = motion_model(mu,w,v,dt);
```
Where is this function located? I might be missing it, but looks like this function was deleted in this commit  a64bc77a0caa541b1cfd0b3947c77d315f5da8af,

majcote commented:
LGTM!

acamalinovic commented:
The function EKF has too much arguments. Instead of using functions such as motion model, sensor model, linearized motion and sensor model, it might pass only their results as a argument of the function. Thus would have as a result  9 arguments.

acamalinovic commented on matlab_simulation/code v1.0/ME597-5-ExtendedKalmanFilter/EKF.m:
```diff
@@ -0,0 +1,24 @@
+    mu = mup + K*(y-sensor_model(mup));
+    S = (eye(n)-K*Ht)*Sp;
+
+end
```
Maybe the number of arguments could be reduced. Instead of using functions such as motion model, sensor model, linearized motion and sensor model as a input arguments, it might pass only their results ? Thus would have as a result 9 arguments or even less.

ajaysinghchhonker replied:
Thanks for the feedback. I'll work on this and update you the changes. \m/

ajaysinghchhonker replied:
No. of argument reduced. Check the new commit. Thanks for your suggestions.

alexsb92 commented on matlab_simulation/code v1.0/ME597-5-ExtendedKalmanFilter/EKF_omnibot_example.m:
```diff
@@ -1,95 +1,89 @@
+    %% Plot results
+    figure(1);clf; hold on;
+    %plot(0,0,'bx', 'MarkerSize', 6, 'LineWidth', 2)
+    %plot([20 -1],[0 0],'b--')
```
I'm not sure if you still need these comments, but i'd be taking them out if not, just so the code is cleaner.

usually for any commented out code that's left in the final version, I put a short reasoning as to why.

ajaysinghchhonker replied:
Oh yeah.. I'll clean them.. Thanks!! :-)

RobWks commented on matlab_simulation/code v1.0/ME597-5-ExtendedKalmanFilter/omnibot_linearize_sensor_model.m:
```diff
@@ -0,0 +1,4 @@
+function Ht = omnibot_derivative_sensor_model(mu)
+    n = length(mu);
+    Ht = eye(n);
+end
```
There is a mismatch here between the name of the file and the name of the function. The function should just be called omnibot_linearize_sensor_model().

ajaysinghchhonker replied:
Yeah.. I forget to commit the changes.. I'll update the changes. Thanks :-)

ajaysinghchhonker replied:
Code updated.

RobWks commented:
I just had to rename that one function where I added the note and it worked for me. The video is a really nice touch, I wonder if there is a way to get the title and axis in the movie? I know you were trying to keep it modular, but I think some of the functions could simply be brought into the EKF function. For instance function 'y' just seems to be doing a transpose?

ajaysinghchhonker committed:
Optimized EKF

No. of argument to EKF function reduced.
EKF function made more generic.
Axis and legend added in video.

ajaysinghchhonker committed:
function name changed

ajaysinghchhonker commented:
Optimized EKF
No. of argument to EKF function reduced.
EKF function made more generic.
Axis and legend added in video.
